arangod/Cluster/MaintenanceFeature.cpp

```diff
@@ -62,7 +62,7 @@ void MaintenanceFeature::init() {
   requiresElevatedPrivileges(false); // ??? this mean admin priv?
 
   // these parameters might be updated by config and/or command line options
-  _maintenanceThreadsMax = static_cast<int32_t>(TRI_numberProcessors() / 4 + 1);
+  _maintenanceThreadsMax = (std::max)(2, static_cast<int32_t>(TRI_numberProcessors() / 4 + 1));
```

Comment:
For info, not casting `2` into `int32_t` here may actually cause trouble on some MacOS platforms.
Because std::max expects the types of its two arguments to be identical, and this is not guaranteed for `2` and `int32_t`.
Can you explicitly cast `2` to an `int32_t` here?